ledger/participant-integration-api/src/main/scala/platform/usermanagement/PersistentUserManagementStore.scala

```diff
@@ -99,30 +100,36 @@ class PersistentUserManagementStore(
     inTransaction(_.getUserInfo) { implicit connection =>
       withUser(id) { dbUser =>
         val rights = backend.getUserRights(internalId = dbUser.internalId)(connection)
-        UserInfo(dbUser.domainUser, rights.map(_.domainRight))
+        val domainUser = toDomainUser(dbUser)
+        UserInfo(domainUser, rights.map(_.domainRight))
       }
     }
   }
 
   override def createUser(
       user: domain.User,
       rights: Set[domain.UserRight],
-  )(implicit loggingContext: LoggingContext): Future[Result[Unit]] = {
+  )(implicit loggingContext: LoggingContext): Future[Result[User]] = {
     inTransaction(_.createUser) { implicit connection: Connection =>
       withoutUser(user.id) {
         val now = epochMicroseconds()
-        val internalId = backend.createUser(user, createdAt = now)(connection)
+        val dbUser = UserManagementStorageBackend.DbUserPayload(
```

Comment:
Don't we need some TODOs around here. This implementation is not finished

- no resourceVersion check/enforcement
- no storage of annotations

Reply:
That's already included in the next PR: https://github.com/digital-asset/daml/pull/14834/files